src/test/java/seedu/address/logic/commands/flashcardcommands/DeleteFlashcardCommandTest.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.flashcardcommands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.commandtestutils.FlashcardBankCommandTestUtil.assertCommandFailure;
+import static seedu.address.logic.commands.commandtestutils.FlashcardBankCommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalFlashcardSets.getTypicalFlashcardBank;
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST;
+import static seedu.address.testutil.TypicalIndexes.INDEX_SECOND;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.model.FlashcardModel;
+import seedu.address.model.FlashcardModelManager;
+import seedu.address.model.flashcard.Flashcard;
+import seedu.address.model.flashcard.FlashcardSet;
+
+public class DeleteFlashcardCommandTest {
+
+    private FlashcardModel model = new FlashcardModelManager(getTypicalFlashcardBank());
+
+    @Test
+    public void execute_validIndexList_success() {
+        FlashcardSet flsetToDeleteFrom = model.getFlashcardSet(INDEX_FIRST);
+        Flashcard flashcardToDelete = model.getFlashcard(flsetToDeleteFrom, INDEX_FIRST);
+
+        DeleteFlashcardCommand deleteCommand = new DeleteFlashcardCommand(INDEX_FIRST, INDEX_FIRST);
+
+        String expectedMessage = String.format(DeleteFlashcardCommand.MESSAGE_DELETE_FLASHCARD_SUCCESS,
+                flashcardToDelete);
+
+        FlashcardModelManager expectedModel = new FlashcardModelManager(getTypicalFlashcardBank());
+        expectedModel.deleteFlashcard(flsetToDeleteFrom, INDEX_SECOND); // There is might be some issue here
```

Comment:
What issue?

Reply:
I'm not sure why during this particular test, the index to be deleted is ` INDEX_SECOND` while it should be `INDEX_FIRST`. This could be a testing issue instead of a bug in the source code, yet I can't find any problem when inspecting both the source code and the testing code. I will check through again and get back after!